internal/services/network/network_packet_capture_resource.go

```diff
@@ -60,6 +60,17 @@ func resourceNetworkPacketCapture() *pluginsdk.Resource {
 				ForceNew: true,
 			},
 
+			"target_type": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(network.PacketCaptureTargetTypeAzureVM),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.PacketCaptureTargetTypeAzureVM),
+					string(network.PacketCaptureTargetTypeAzureVMSS),
+				}, false),
+			},
```

Comment:
We can infer the target type from the `target_resource_id`, so why do we need to specify this at all?

Reply:
I'm assuming further targets that _may_ be supported will still require specifying the id for said target, in which case the target type should still be inferrable? So I fail to see how optionally supplying a `target_type` adds any benefit to the end user at the moment or improves maintainability.

Reply:
After some further thinking on this, this resource needs to be split into two separate ones that support VMs and VMSSs individually since this allows us to provide a better user experience by only exposing fields which are relevant for VM and VMSS. 

So what we need are the following new resources where the `include_instance_ids` and `exclude_instance_ids` can become top level fields within the resource
* `azurerm_virtual_machine_packet_capture`
* `azurerm_virtual_machine_scale_set_packet_capture`

And to deprecate this resource `azurerm_network_packet_capture` and remove it in 4.0.

Reply:
@stephybun , I intend to update this PR to infer the target_type. For splitting the resource, I think we could implement it in another new PR. Make sense?